lesson11/js/main.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+const menuContainer = document.getElementById("js-menu_container");
+const menuLists = document.createElement("ul");
+const loading = document.getElementById("js-loading");
+
+function showLoadingImg() {
+  const loadingImg = document.createElement("img");
+  loadingImg.src = "images/loading-circle.gif";
+  loadingImg.classList.add("el_loadingImg");
+
+  loading.appendChild(loadingImg);
+}
+
+function createMenuList({data}) {
+  const fragment = document.createDocumentFragment();
+
+  data.forEach((value) => {
+      const listItem = document.createElement("li");
+  
+      const listImg = document.createElement("img");
+      listImg.src = value.img;
+      listImg.alt = value.alt;
+      listImg.width = 40;
+      listImg.height = 40;
+  
+      const listAnchor = document.createElement("a");
+      listAnchor.href = value.a;
+      listAnchor.textContent = value.text;
+  
+      fragment.appendChild(listItem).appendChild(listAnchor).appendChild(listImg);
+
+    });
+
+  menuContainer.appendChild(menuLists).appendChild(fragment);
+}
+
+async function fetchMenuContents(){
+  showLoadingImg();
+
+  const menuDataURL = 'https://jsondata.okiba.me/v1/json/Aticp210816214718';
+  const response = await fetch(menuDataURL);
+  
+  if(response.status === 200){
+    return await response.json();
+
+  }else{
+   console.error("Could not get the value");
+  }
```

Comment:
Try to use try-catch instead of if-else!

Reply:
Thank you for review!

I rewrote it to try-catch.
However, I'm worried about how to write the catch error code.
I would like to seek your advice on this matter.
Do I have to write an if statement with try here and pass the status code in case of an error?
I would appreciate your help.

Reply:
```
  try {
    const response = await fetch(MENU_DATA_URL);
    const json = await response.json();
    return json.data;
  } catch {
    console.error("error： HTTP error, status =" + response.status);
  }
```

It is done this way, but I think you are getting an error as `response` is `undefined`.
The reason for this is that the response is in the `try block`, and from the `catch block`, the `response` is out of scope.
Therefore, it is not clear at this point whether there is a way to pass the status code here, and I don't think it is that important.
I don't think it is that important, because the status code can be checked by the validation tool.
I think once we have it, we can just write a string that tells us that an error has occurred.

Reply:
Thank you for review.

> It is done this way, but I think you are getting an error as response is undefined.
I understand that the constant responsen is out of scope for catch.

> I don't think it is that important, because the status code can be checked by the validation tool.
As Sho-san said, I didn't think I needed a status code here, so I made it simple to display the text.